src/sonic-p4rt/Makefile

```diff
@@ -0,0 +1,47 @@
+.ONESHELL:
+SHELL = /bin/bash
+.SHELLFLAGS += -e
+
+BAZEL_TARGET_PATH = p4rt_app
+BAZEL_BUILD_TARGETS = //p4rt_app:p4rt_deb //p4rt_app:p4rt_dbg_deb
+
+# Enable debug symbols for remote debugging (generate a .dwp).
+BAZEL_BUILD_OPTS += --fission=yes --features=per_object_debug_info
+
+# Override sonic_swss_common to use the version specific to this buildimage.
+BAZEL_BUILD_OPTS += --override_repository=sonic_swss_common=/sonic/src/sonic-swss-common
+
+# Build optimized, stripped binaries.
+BAZEL_BUILD_OPTS += -c opt
+
+# Use a host directory to store Bazel cache, if mounted. This will speed up
+# incremental rebuilds on the same system for developers that makes changes
+# to sonic-pins code.
+#
+# To build with Bazel cache, add DOCKER_BUILDER_USER_MOUNT:
+#   DOCKER_BUILDER_USER_MOUNT=<directory on host system>:$BAZEL_CACHE:rw
+#
+# For example:
+#   DOCKER_BUILDER_USER_MOUNT=/tmp/bazel_cache:/bazel:rw make target/...
+#
+BAZEL_CACHE ?= /bazel
+BAZEL_OPTS += $(shell test -d $(BAZEL_CACHE) && echo --output_user_root=$(BAZEL_CACHE)/cache)
+
+MAIN_TARGET = $(SONIC_P4RT)
+DERIVED_TARGETS = $(SONIC_P4RT_DBG)
+
+$(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
+	function cleanup {
+		# Note: make seems to hang if Bazel is still running
+		cd $(CURDIR)/sonic-pins && bazel $(BAZEL_OPTS) shutdown
+	}
+	trap cleanup EXIT
+	pushd ./sonic-pins
```

Comment:
```suggestion
	mv ./sonic-pins
```

Reply:
@rhalstea Sorry, I didn’t mean `mv` here, I meant `cd`. Oops!

I don’t know if Make cares that we change the working directory during the build. If not, then `cd` should work in both places. If Make does care, then we can use `pushd` for the first one and `cd` for the second. I can test this and update accordingly.

Reply:
Lots of other Makefiles use pushd, so will keep that convention here as well.